eng/native/configurecompiler.cmake

```diff
@@ -453,6 +453,14 @@ if (CLR_CMAKE_HOST_UNIX)
     add_compile_options(-Wno-incompatible-ms-struct)
 
     add_compile_options(-Wno-reserved-identifier)
+
+    # clang 16.0 introduced buffer hardening https://discourse.llvm.org/t/rfc-c-buffer-hardening/65734
+    # which we are not conforming to yet.
+    add_compile_options(-Wno-unsafe-buffer-usage)
```

Comment:
@AaronRobinsonMSFT, this is very "new", but it was even showing violations in safecrt.h 😁 maybe something we can/should audit.

Reply:
Awesome. Nice catch @am11. @RobertHenry6bev is also interested in rooting out these sorts of problems. File an issue?

Reply:
https://github.com/dotnet/runtime/issues/81577